python/config/series_description_matching.py

```diff
@@ -0,0 +1,180 @@
+    match_data = {}
+    tcs_with_duplicate_keys = []
+    for tc in aspace_containers:
+        tc_indicator_with_series = tc.get("indicator")
+        # if indicator begins with a number, split it into indicator and series
+        # numerical part is the indicator, the rest is the series
+        # e.g. "330M" -> indicator "330", series "M"
+        if tc_indicator_with_series[0].isdigit():
+            tc_indicator = ""
+            for char in tc_indicator_with_series:
+                if char.isdigit():
+                    tc_indicator += char
+                else:
+                    break
+            tc_series = tc_indicator_with_series[len(tc_indicator) :]
+        # if indicator does not begin with a number, also split it into indicator and series
+        # hyphen delimited
+        # e.g. "SR-130" -> indicator "130", series "SR"
+        else:
+            tc_indicator = tc_indicator_with_series.split("-")[1]
+            tc_series = tc_indicator_with_series.split("-")[0]
+        tc_type = tc.get("type")
+        # double check for duplicates
+        if (tc_indicator, tc_type, tc_series) in match_data:
+            if logger:
+                logger.error(
+                    f"Duplicate top container found:"
+                    f" {tc_indicator} {tc_type} {tc_series} {tc.get('uri')}."
+                    f" Existing top container:"
+                    f" {match_data[(tc_indicator, tc_type, tc_series)].get('uri')}."
+                    " Skipping both top containers."
+                )
+            tcs_with_duplicate_keys.append(
+                (tc.get("uri"), tc_indicator, tc_type, tc_series)
+            )
+            tcs_with_duplicate_keys.append(
+                (
+                    match_data[(tc_indicator, tc_type, tc_series)].get("uri"),
+                    tc_indicator,
+                    tc_type,
+                    tc_series,
+                )
+            )
+            # remove the duplicate
+            del match_data[(tc_indicator, tc_type, tc_series)]
+            # skip this top container
+            continue
+        match_data[(tc_indicator, tc_type, tc_series)] = tc
+    return match_data, tcs_with_duplicate_keys
+
+
+def _get_alma_match_data(
+    alma_items: list, logger: Optional[Any] = None
+) -> tuple[dict[tuple], list[tuple]]:
+    """Parses Alma item descriptions into container type, indicator, and series
+    and normalizes the indicator by removing leading zeroes and trailing " RESTRICTED".
+    Returns a dictionary with the normalized indicator, type, and series as keys, and
+    a list of items with duplicate keys.
+    """
+    match_data = {}
+    items_with_duplicate_keys = []
+    for item in alma_items:
+        description = item.get("description")
+        # split description into series and container type/indicator (space and period delimited)
+        # e.g. "ser.P box.0011" -> "P", "box", "0011"
+        alma_series = description.split(" ")[0].split(".")[1]
```

Comment:
Since the Alma descriptions appear to be uniform, I think this code is fine as is.  It's clearly commented and doesn't lend itself to "cleverness" - no easy trick (that I see) to grab the 3 elements via a single operation.